atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -153,7 +158,8 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         checkOpen();
         try {
             TransactionAndImmutableTsLock txAndLock =
-                    runTimed(() -> setupRunTaskWithConditionThrowOnConflict(condition), "setupTask");
+                    runTimed(() -> Iterables.getOnlyElement(
+                            setupRunTaskBatchWithConditionThrowOnConflict(ImmutableList.of(condition))), "setupTask");
```

Comment:
I quite like this, it's just an unwrapping and you're done. You could have a private method for setup*, but I'm not fussed either way.